hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsFastpathConnection.java

```diff
@@ -0,0 +1,211 @@
+ */
+public class AbfsFastpathConnection extends AbfsHttpOperation {
+
+  int defaultTimeout = Integer.valueOf(DEFAULT_TIMEOUT);
+  String fastpathFileHandle;
+  FastpathResponse response = null;
+
+  public String getFastpathFileHandle() {
+    return fastpathFileHandle;
+  }
+
+  public AbfsFastpathConnection(final AbfsRestOperationType opType,
+      final URL url,
+      final String method,
+      final AuthType authType,
+      final String authToken,
+      List<AbfsHttpHeader> requestHeaders,
+      final String fastpathFileHandle) throws IOException {
+    super(opType, url, method, authType, authToken, requestHeaders);
+    this.authType = authType;
+    this.authToken = authToken;
+    this.fastpathFileHandle = fastpathFileHandle;
+    this.requestHeaders = requestHeaders;
+  }
+
+  public String getResponseHeader(String httpHeader) {
+    return response.getResponseHeaders().get(httpHeader);
+  }
+
+  public Map<String, List<String>> getRequestHeaders() {
+    Map<String, List<String>> headers
+        = new HashMap<String, java.util.List<String>>();
+    for (AbfsHttpHeader abfsHeader : this.requestHeaders) {
+      headers.put(abfsHeader.getName(),
+          Collections.singletonList(abfsHeader.getValue()));
+    }
+
+    headers.put(
+        X_MS_CLIENT_REQUEST_ID,
+        Collections.singletonList(this.clientRequestId));
+
+    return headers;
+  }
+
+  public String getRequestHeader(String header) {
+    String value ="";
+    for (AbfsHttpHeader abfsHeader : this.requestHeaders) {
+      if (abfsHeader.getName().equals(header)) {
+        value = abfsHeader.getValue();
+        break;
+      }
+    }
+
+    return value;
+  }
+
+  /**
+   * Gets and processes the HTTP response.
+   *
+   * @param buffer a buffer to hold the response entity body
+   * @param offset an offset in the buffer where the data will being.
+   * @param length the number of bytes to be written to the buffer.
+   *
+   * @throws IOException if an error occurs.
+   */
+  public void processResponse(byte[] buffer, final int offset, final int length) throws IOException {
+    switch (this.opType) {
+    case FastpathOpen:
+      long startTime = System.nanoTime();
+      processFastpathOpenResponse();
+      this.recvResponseTimeMs = elapsedTimeMs(startTime);
+      break;
+    case FastpathRead:
+      startTime = System.nanoTime();
+      processFastpathReadResponse(buffer, offset, length);
+      this.recvResponseTimeMs = elapsedTimeMs(startTime);
+      break;
+    case FastpathClose:
+      startTime = System.nanoTime();
+      processFastpathCloseResponse();
+      this.recvResponseTimeMs = elapsedTimeMs(startTime);
+      break;
+    default:
+      throw new FastpathException("Invalid state");
+    }
+  }
+
+  private void setStatusFromFastpathResponse(FastpathResponse response) {
+    this.response = response;
+    this.statusCode = response.getHttpStatus();
+    this.statusDescription = String.valueOf(response.getHttpStatus());
```

Comment:
is description meant to store a String version of the http status (int)?

Reply:
Supposed to be Http status code's description. But will have to introduce a dependency just to get that done. Will keep the description to the store error description that we get from fastpath.